c_api/tests/c_api_behavior_test.cpp

```diff
@@ -113,3 +113,504 @@ TEST_F(CapiTest, TestBehaviorGetRuntimeCapabilities) {
+      }
+
+      // runtime and allocate_info are both null
+      {
+        ti_allocate_memory(TI_NULL_HANDLE, nullptr);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+
+      // runtime is not null, allocate_info is null
+      {
+        ti_allocate_memory(runtime, nullptr);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+
+      // runtime is null, allocate is not null;
+      {
+        TiMemoryAllocateInfo allocate_info;
+        allocate_info.size = 1024;
+        ti_allocate_memory(TI_NULL_HANDLE, &allocate_info);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+
+      ti_destroy_runtime(runtime);
+    }
+  };
+
+  inner(TI_ARCH_VULKAN);
+}
+
+TEST_F(CapiTest, TestBehaviorFreeMemory) {
+  auto inner = [](TiArch arch) {
+    if (ti::is_arch_available(arch)) {
+      {
+        TiRuntime runtime = ti_create_runtime(arch);
+        TiMemoryAllocateInfo *allocate_info = new TiMemoryAllocateInfo;
+        allocate_info->size = 1024;
+        allocate_info->usage = TI_MEMORY_USAGE_STORAGE_BIT;
+        TiMemory memory = ti_allocate_memory(runtime, allocate_info);
+        ti_free_memory(runtime, memory);
+        CHECK_TAICHI_SUCCESS();
+      }
+      // runtime & allocate_info are both null
+      {
+        ti_free_memory(TI_NULL_HANDLE, nullptr);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+
+      // runtime is null and allocate_info is valid
+      {
+        TiRuntime runtime = ti_create_runtime(TI_ARCH_VULKAN);
+        TiMemoryAllocateInfo allocate_info;
+        allocate_info.size = 1024;
+        allocate_info.usage = TI_MEMORY_USAGE_STORAGE_BIT;
+        TiMemory memory = ti_allocate_memory(runtime, &allocate_info);
+        ti_free_memory(TI_NULL_HANDLE, memory);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+
+      // runtime is not null and allocate_info is null
+      {
+        TiRuntime runtime = ti_create_runtime(arch);
+        ti_free_memory(runtime, nullptr);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+    }
+  };
+  inner(TI_ARCH_VULKAN);
+}
+
+TEST_F(CapiTest, TestBehaviorMapMemory) {
+  TiMemoryAllocateInfo allocate_info;
+  allocate_info.size = 1024;
+  allocate_info.usage = TI_MEMORY_USAGE_STORAGE_BIT;
+
+  auto inner = [&](TiArch arch) {
+    if (ti::is_arch_available(arch)) {
+      {
+        TiRuntime runtime = ti_create_runtime(arch);
+        TiMemory memory = ti_allocate_memory(runtime, &allocate_info);
+        TI_ASSERT(memory != TI_NULL_HANDLE);
+        CHECK_TAICHI_SUCCESS();
+        ti_map_memory(runtime, memory);
+        CHECK_TAICHI_SUCCESS();
+        ti_unmap_memory(runtime, memory);
+      }
+
+      // runtime & memory are both null
+      {
+        ti_map_memory(TI_NULL_HANDLE, TI_NULL_HANDLE);
+        CHECK_TAICHI_ERROR_IS(TI_ERROR_ARGUMENT_NULL);
+      }
+
+      // runtime is null, memory is valid
+      {
+        TiRuntime runtime = ti_create_runtime(arch);
```

Comment:
please ensure everything you create or allocate is destroyed. we want to keep the CI stable. Or just use something that automatically does the destroy in its destructor like `ti::Runtime`, if the object itself (`TiRuntime` in this case) is not the tested thing.